features/conversation/conversation-message/conversation-message-reactions/conversation-message-reaction-drawer/conversation-message-reaction-drawer.tsx

```diff
@@ -1,142 +1,193 @@
 import Avatar from "@components/Avatar";
 import { BottomSheetContentContainer } from "@design-system/BottomSheet/BottomSheetContentContainer";
 import { BottomSheetHeader } from "@design-system/BottomSheet/BottomSheetHeader";
-import { BottomSheetModal } from "@design-system/BottomSheet/BottomSheetModal";
+import { CustomBottomSheet } from "@design-system/BottomSheet/BottomSheet";
 import { HStack } from "@design-system/HStack";
 import { Text } from "@design-system/Text";
 import { BottomSheetScrollView } from "@gorhom/bottom-sheet";
 import { FlashList } from "@shopify/flash-list";
 import { ThemedStyle, useAppTheme } from "@theme/useAppTheme";
-import { memo, useCallback, useState } from "react";
-import { TextStyle, TouchableHighlight, ViewStyle } from "react-native";
+import { memo, useCallback, useState, useRef, useEffect } from "react";
+import {
+  TextStyle,
+  TouchableHighlight,
+  ViewStyle,
+  Modal,
+  Platform,
+} from "react-native";
 import { ScrollView } from "react-native-gesture-handler";
 import { useSafeAreaInsets } from "react-native-safe-area-context";
+import Animated, { FadeIn, FadeOut } from "react-native-reanimated";
+import { BottomSheetMethods } from "@gorhom/bottom-sheet/lib/typescript/types";
 
 import {
-  bottomSheetModalRef,
-  resetMessageReactionsDrawer,
+  closeMessageReactionsDrawer,
   useMessageReactionsRolledUpReactions,
+  useMessageReactionsDrawerVisible,
 } from "./conversation-message-reaction-drawer.service";
 
 export const MessageReactionsDrawer = memo(function MessageReactionsDrawer() {
   const { theme, themed } = useAppTheme();
   const insets = useSafeAreaInsets();
   const rolledUpReactions = useMessageReactionsRolledUpReactions();
+  const isVisible = useMessageReactionsDrawerVisible();
+  const bottomSheetRef = useRef<BottomSheetMethods>(null);
+  const [filterReactions, setFilterReactions] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (isVisible) {
+      bottomSheetRef.current?.snapToIndex(0);
+      setFilterReactions(null);
+    }
+  }, [isVisible]);
 
   const handleDismiss = useCallback(() => {
-    resetMessageReactionsDrawer();
+    bottomSheetRef.current?.close();
     setFilterReactions(null);
   }, []);
 
-  // State for managing the active filter, e.g., ❤️, 👍, etc.
-  const [filterReactions, setFilterReactions] = useState<string | null>(null);
+  if (!isVisible) {
+    return null;
+  }
 
   return (
-    <BottomSheetModal
-      onDismiss={handleDismiss}
-      ref={bottomSheetModalRef}
-      topInset={insets.top}
-      snapPoints={["50%", "100%"]}
+    <Modal
+      visible={isVisible}
```

Comment:
Doesn'T make any sense because you have

 if (!isVisible) {
     return null;
   }

Above